shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -237,6 +237,11 @@ - (void)loadView {
 #pragma mark - Managing launch views
 
 - (void)installSplashScreenViewIfNecessary {
+  // If the engine wasn't created by this FlutterViewController, we should avoid installing
```

Comment:
The problem seems to be that the the callback is never getting invoked in this case.  It might be better to fix that than what I'm proposing here though.

Reply:
Making a little progress: `installSplashScreenViewIfNecessary` gets called before the user gets a chance to actually set the splashscreen.   The user then tries to set a splash screen, which is messing something up.